In [27]:
import pandas as pd

In [128]:
df = pd.read_csv('reservations_stats.csv', skiprows=13, sep=';')
df.head()

,Month,Total reservations,Confirmed reservations,Option reservations,Offer reservations,No-show reservations,Cancelled reservations,No-show rate (%),Cancellation rate (%),Booking window (days)
0,September 2025,242,198,0,0,3,41,1.24,16.94,11.59
1,October 2025,254,194,0,0,0,60,0,23.62,13.66
2,November 2025,240,207,0,0,0,33,0,13.75,6.86
3,December 2025,322,252,0,0,2,68,0.62,21.12,20.65
4,January 2026,232,189,0,0,0,43,0,18.53,9.02


In [137]:
df[['Month',
    'Total reservations',
    'Confirmed reservations',
    'Cancelled reservations',
    'Cancellation rate (%)',
    'Booking window (days)']]

,Month,Total reservations,Confirmed reservations,Cancelled reservations,Cancellation rate (%),Booking window (days)
0,September 2025,242,198,41,16.94,11.59
1,October 2025,254,194,60,23.62,13.66
2,November 2025,240,207,33,13.75,6.86
3,December 2025,322,252,68,21.12,20.65
4,January 2026,232,189,43,18.53,9.02
5,February 2026,248,203,44,17.74,6.12
6,March 2026,117,50,67,57.26,26.41
7,April 2026,132,106,26,19.7,15.44
8,May 2026,225,172,52,23.11,12.4
9,June 2026,221,181,39,17.65,7.16


In [145]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 10 columns):
 #   Column                  Non-Null Count  Dtype
---  ------                  --------------  -----
 0   Month                   14 non-null     str  
 1   Total reservations      14 non-null     str  
 2   Confirmed reservations  14 non-null     str  
 3   Option reservations     14 non-null     str  
 4   Offer reservations      14 non-null     str  
 5   No-show reservations    14 non-null     str  
 6   Cancelled reservations  14 non-null     str  
 7   No-show rate (%)        14 non-null     str  
 8   Cancellation rate (%)   14 non-null     str  
 9   Booking window (days)   14 non-null     str  
dtypes: str(10)
memory usage: 1.3 KB


In [146]:
# df.columns

An analysis of bookings for the period from September 2025 to August 2026 reveals a pronounced seasonality. Over the course of the year, 2,824 bookings were recorded, of which 2,223 were confirmed and 592 were canceled. The average cancellation rate was 20.96%, and the average booking window was 12.23 days. The highest volume of bookings was observed in August 2026, while March 2026 stands out as an anomalous month: the number of bookings dropped sharply, and the cancellation rate reached 57.26%. This indicates the need for a separate analysis of the reasons for the March deviation.

In [ ]:
df = pd.read_csv('saleschannels_stats.csv')
df.head()

In [157]:
with open('saleschannels_stats.csv', 'r', encoding='utf-8') as file:
    for i in range(20):
        print(i, file.readline().strip())

0 "Statistics from 01/09/2025 until 31/08/2026"
1 ""
2 "Reservation status: Confirmed, Option (non-expired)"
3 "Board(s): any"
4 "Reservation tag(s): any"
5 ""
6 "Room type(s): any"
7 "Room tag(s): any"
8 ""
9 "Sales channel(s): any"
10 "Agencies: any"
11 "Corporates: any"
12 ""
13 "Sales channel";"Number of reservations";"Rooms revenue";"Total revenue";"Sold rooms";"ADR";"Reservation Rate (all / same sales channel)";"Reservation Rate (same channel type)"
14 "All reservations";"2223";"1,449,813.02 ILS";"1,577,461.45 ILS";"3366";"430.72 ILS";"-"
15 "Manual reservations";"873";"536,014.11 ILS";"577,919.11 ILS";"1313";"408.24 ILS";"39.27%"
16 "Zak Online Reception";"0";"0 ILS";"0 ILS";"0";"0 ILS";"0%"
17 "Zak Channel Manager";"1350";"913,798.91 ILS";"999,542.34 ILS";"2053";"445.10 ILS";"60.73%"
18 
19 "Direct reservations";"873";"536,014.11 ILS";"577,919.11 ILS";"1313";"408.24 ILS";"-"


In [165]:
df = pd.read_csv('saleschannels_stats.csv',sep=';',skiprows=13)
df.head()

,Sales channel,Number of reservations,Rooms revenue,Total revenue,Sold rooms,ADR,Reservation Rate (all / same sales channel),Reservation Rate (same channel type)
0,All reservations,2223,"1,449,813.02 ILS","1,577,461.45 ILS",3366,430.72 ILS,-,NaN
1,Manual reservations,873,"536,014.11 ILS","577,919.11 ILS",1313,408.24 ILS,39.27%,NaN
2,Zak Online Reception,0,0 ILS,0 ILS,0,0 ILS,0%,NaN
3,Zak Channel Manager,1350,"913,798.91 ILS","999,542.34 ILS",2053,445.10 ILS,60.73%,NaN
4,Direct reservations,873,"536,014.11 ILS","577,919.11 ILS",1313,408.24 ILS,-,NaN


In [166]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 21 entries, 0 to 20
Data columns (total 8 columns):
 #   Column                                       Non-Null Count  Dtype  
---  ------                                       --------------  -----  
 0   Sales channel                                21 non-null     str    
 1   Number of reservations                       21 non-null     int64  
 2   Rooms revenue                                21 non-null     str    
 3   Total revenue                                21 non-null     str    
 4   Sold rooms                                   21 non-null     int64  
 5   ADR                                          21 non-null     str    
 6   Reservation Rate (all / same sales channel)  21 non-null     str    
 7   Reservation Rate (same channel type)         0 non-null      float64
dtypes: float64(1), int64(2), str(5)
memory usage: 1.4 KB


In [167]:
df[['Sales channel',
    'Number of reservations',
    'Rooms revenue',
    'Sold rooms',
    'ADR',
    'Reservation Rate (all / same sales channel)']]

,Sales channel,Number of reservations,Rooms revenue,Sold rooms,ADR,Reservation Rate (all / same sales channel)
0,All reservations,2223,"1,449,813.02 ILS",3366,430.72 ILS,-
1,Manual reservations,873,"536,014.11 ILS",1313,408.24 ILS,39.27%
2,Zak Online Reception,0,0 ILS,0,0 ILS,0%
3,Zak Channel Manager,1350,"913,798.91 ILS",2053,445.10 ILS,60.73%
4,Direct reservations,873,"536,014.11 ILS",1313,408.24 ILS,-
5,YouBook,95,"58,882 ILS",155,379.88 ILS,10.88%
6,zCrm,0,0 ILS,0,0 ILS,0%
7,Fast Planner,725,"441,911.11 ILS",1064,415.33 ILS,83.05%
8,Copied,45,"30,737 ILS",82,374.84 ILS,5.15%
9,Mobile App,8,"4,484 ILS",12,373.67 ILS,0.92%


In [168]:
df[['Sales channel', 'Number of reservations', 'ADR']]

,Sales channel,Number of reservations,ADR
0,All reservations,2223,430.72 ILS
1,Manual reservations,873,408.24 ILS
2,Zak Online Reception,0,0 ILS
3,Zak Channel Manager,1350,445.10 ILS
4,Direct reservations,873,408.24 ILS
5,YouBook,95,379.88 ILS
6,zCrm,0,0 ILS
7,Fast Planner,725,415.33 ILS
8,Copied,45,374.84 ILS
9,Mobile App,8,373.67 ILS


In [169]:
channel_data = df[df['Sales channel'].isin([
    'Manual reservations',
    'Booking',
    'Expedia',
    'Agoda',
    'WooDoo'
])]

channel_data[['Sales channel', 'Number of reservations', 'ADR']]

,Sales channel,Number of reservations,ADR
1,Manual reservations,873,408.24 ILS
13,Agoda,22,306.71 ILS
15,Booking,1102,444.85 ILS
17,Expedia,52,389.58 ILS
19,WooDoo,174,488.59 ILS


Booking              49.57%
Manual               39.27%
WooDoo / Simplex      7.83%
Expedia               2.34%
Agoda                 0.99%

An analysis of sales channels shows that Booking is the primary source of bookings, accounting for almost half of all confirmed bookings. However, approximately 39% of bookings are manually generated by hotel staff. The hotel's own website, which is managed through WooDoo/Simplex, accounts for approximately 8% of bookings but has the highest ADR—488.59 ILS. This suggests potential for developing direct sales through the website.

In [173]:
# df = pd.read_csv('revenue_stats.csv')


In [174]:
with open('revenue_stats.csv', 'r', encoding='utf-8') as file:
    for i in range(40):
        print(i, file.readline().strip())

0 "Statistics from 01/09/2025 until 31/08/2026"
1 ""
2 "Reservation status: Confirmed, Option (non-expired)"
3 "Board(s): any"
4 "Reservation tag(s): any"
5 ""
6 "Room type(s): any"
7 "Room tag(s): any"
8 ""
9 "Sales channel(s): any"
10 "Agencies: any"
11 "Corporates: any"
12 ""
13 "Month";"Total Revenue (ILS)";"Rooms Revenue (ILS)";"Meals Revenue (ILS)";"Extras Revenue (ILS)";"Rooms Revenue rate (%)";"Available rooms";"Closed rooms";"Sold rooms";"Occupancy rate (%)";"RevPAR (ILS)";"ADR (ILS)";"TrevPAR (ILS)";"TrevPOR (ILS)";"Sold rooms (BUDG)";"Sold rooms (CLAS)";"Sold rooms (DLXA)";"Sold rooms (ECON)";"Sold rooms (FMLA)";"Sold rooms (FMLB)";"Sold rooms (SPCV)";"Sold rooms (SPSV)";"Sold rooms (day)"
14 "September 2025";"120419.33";"112939.33";"5315";"2165";"93.79";"780";"0";"272";"34.87";"144.79";"415.22";"154.38";"442.72";"59";"55";"10";"45";"21";"11";"32";"31";"8"
15 "October 2025";"148799.23";"134804.23";"10605";"3390";"90.59";"806";"0";"301";"37.34";"167.25";"447.85";"184.61";"49

In [175]:
df = pd.read_csv( 'revenue_stats.csv',sep=';',skiprows=13,nrows=12)
df.head()

,Month,Total Revenue (ILS),Rooms Revenue (ILS),Meals Revenue (ILS),Extras Revenue (ILS),Rooms Revenue rate (%),Available rooms,Closed rooms,Sold rooms,Occupancy rate (%),...,TrevPOR (ILS),Sold rooms (BUDG),Sold rooms (CLAS),Sold rooms (DLXA),Sold rooms (ECON),Sold rooms (FMLA),Sold rooms (FMLB),Sold rooms (SPCV),Sold rooms (SPSV),Sold rooms (day)
0,September 2025,120419.33,112939.33,5315,2165.00,93.79,780,0,272,34.87,...,442.72,59,55,10,45,21,11,32,31,8
1,October 2025,148799.23,134804.23,10605,3390.00,90.59,806,0,301,37.34,...,494.35,41,71,17,57,26,18,28,32,11
2,November 2025,129406.03,114592.03,11730,3084.00,88.55,780,0,293,37.56,...,441.66,54,76,8,51,25,15,28,22,14
3,December 2025,192408.54,174588.25,14720,3100.29,90.74,806,0,375,46.53,...,513.09,58,79,19,67,32,26,34,36,24
4,January 2026,105658.42,97310.42,5820,2528.00,92.10,806,0,248,30.77,...,426.04,55,51,9,48,11,9,31,25,9


In [176]:
df[[
    'Month',
    'Total Revenue (ILS)',
    'Rooms Revenue (ILS)',
    'Sold rooms',
    'Occupancy rate (%)',
    'RevPAR (ILS)',
    'ADR (ILS)',
    'TrevPAR (ILS)'
]]

,Month,Total Revenue (ILS),Rooms Revenue (ILS),Sold rooms,Occupancy rate (%),RevPAR (ILS),ADR (ILS),TrevPAR (ILS)
0,September 2025,120419.33,112939.33,272,34.87,144.79,415.22,154.38
1,October 2025,148799.23,134804.23,301,37.34,167.25,447.85,184.61
2,November 2025,129406.03,114592.03,293,37.56,146.91,391.10,165.91
3,December 2025,192408.54,174588.25,375,46.53,216.61,465.57,238.72
4,January 2026,105658.42,97310.42,248,30.77,120.73,392.38,131.09
5,February 2026,126278.84,117626.84,297,40.80,161.58,396.05,173.46
6,March 2026,31865.40,30885.40,74,9.18,38.32,417.37,39.54
7,April 2026,55509.38,53014.38,129,16.54,67.97,410.96,71.17
8,May 2026,140746.42,131037.28,301,37.34,162.58,435.34,174.62
9,June 2026,157847.09,144367.09,325,41.67,185.09,444.21,202.37


An analysis of revenue and hotel KPIs shows that in typical months, hotel performance is determined by a combination of occupancy and average room rate. The strongest results are observed in December 2025, July, and especially August 2026. In August, occupancy reached 51.24%, ADR was 461.55 ILS, and RevPAR was 236.52 ILS, making it one of the best-performing months during the period under review. March 2026 should be considered separately, as the sharp drop in occupancy and the high number of cancellations were caused by an unstable external environment rather than typical market factors. April also partially reflects the effects of this period.

In [184]:
df = pd.read_csv(
    'occupancy_stats.csv',
    sep=';',
    skiprows=13
)

df.head()

,Month,Daily arrivals,Sold occupancy,Available occupancy,Occupancy rate (%),Average room occupancy,Average length of stay,RevPOB (ILS),RevPAB (ILS),N. unique guests
0,September 2025,400,503,1560,32.24,1.85,1.26,224.53,72.40,192
1,October 2025,451,606,1612,37.59,2.01,1.34,222.45,83.63,193
2,November 2025,423,522,1560,33.46,1.78,1.23,219.52,73.46,225
3,December 2025,571,752,1612,46.65,2.01,1.32,232.17,108.31,256
4,January 2026,351,446,1612,27.67,1.8,1.27,218.18,60.37,187


In [185]:
df.columns

Index(['Month', 'Daily arrivals', 'Sold occupancy', 'Available occupancy',
       'Occupancy rate (%)', 'Average room occupancy',
       'Average length of stay', 'RevPOB (ILS)', 'RevPAB (ILS)',
       'N. unique guests'],
      dtype='str')

In [187]:
occupancy_analysis = df[[
    'Month',
    'Sold occupancy',
    'Available occupancy',
    'Occupancy rate (%)',
    'Average room occupancy',
    'Average length of stay',
    'N. unique guests'
]]
occupancy_analysis

,Month,Sold occupancy,Available occupancy,Occupancy rate (%),Average room occupancy,Average length of stay,N. unique guests
0,September 2025,503,1560,32.24,1.85,1.26,192
1,October 2025,606,1612,37.59,2.01,1.34,193
2,November 2025,522,1560,33.46,1.78,1.23,225
3,December 2025,752,1612,46.65,2.01,1.32,256
4,January 2026,446,1612,27.67,1.8,1.27,187
5,February 2026,538,1456,36.95,1.81,1.4,194
6,March 2026,125,1612,7.75,1.69,1.52,45
7,April 2026,226,1560,14.49,1.75,1.17,94
8,May 2026,527,1612,32.69,1.75,1.51,173
9,June 2026,562,1560,36.03,1.73,1.49,192


Occupancy analysis reveals a pronounced seasonality in the hotel's operations. The highest occupancy rates were observed in August 2026 (53.11%) and December 2025 (46.65%). In most other months, occupancy rates ranged between 30% and 40%. The average length of stay is approximately 1–1.5 nights, indicating a predominance of short-term trips. The average number of guests per occupied room is approximately two. March and part of April 2026 should be considered separately, as the figures for this period were significantly distorted by the unstable external environment.

Less dependence on OTAs, more clear direct channels, better source capture for manual bookings, dynamic price management by season, and separate cancellation controls.